### ISSUE TITLE
Better highlight with chosen colorscheme?

### ISSUE THREAD

**Issue by lkhphuc** [wookayin/semshi#16]
Hi, I just want to put this out here as a potential discussion.

Currently Semshi applies some certain highlighting colors irrespective of the colorscheme. With the rise of LSP Semantic highlight, there are many colorscheme out there with dedicated support for this.

Should Semshi adapts to the colorscheme where possible?

An example is my contribution of Semshi extra to LazyVim, where I only applied style, not color, or linked to LSP color provided by the colorscheme. https://github.com/LazyVim/LazyVim/pull/1149/files

**Comment by wookayin** [comment]
Yes, semshi was written a long time ago when LSP and treesitter was not landed yet. Semshi's highlight setup is `hi def` so it won't override any existing configs if any. 

In the next version, I'm going to change the default highlight linked to @lsp like highlight groups.
